website_sale_exception/models/sale.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2020 ADHOC
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    @api.multi
+    def action_confirm(self):
+        if self.detect_exceptions():
```

Comment:
Isn't this modifying also the behavior of regular sales orders?

Reply:
it's filtering on orders with website_id right after that, so I would say it's not

Reply:
OK, I see, but shouldn't this be done after calling super then?

Reply:
I did it for some reason but at this moment do not remember, I will try to do it after the super() and test it. Let you know the result

Reply:
@pedrobaeza 

I have check the behavior: If we do the super first, then the website sale orders that has detect exceptions will be pass to sent state but they will not reflect the exception info in the sale order, we will never be inform that an exception occurs.

![image](https://user-images.githubusercontent.com/7593953/81734629-2cf63a80-946a-11ea-8a9f-8b1f5978ba93.png)

The way we are doing it now let us to detect exceptions, pass those sale orders to sent state, and then run the normal exception workflow so this way we will have a sent sale order but we will have the info about the exception that occurs trying to validate the order.

![image](https://user-images.githubusercontent.com/7593953/81734643-32ec1b80-946a-11ea-97c0-cd5e233718aa.png)

Reply:
OK, thanks for the explanations!